torch/_dynamo/variables/builder.py

```diff
@@ -550,40 +550,65 @@ def wrap_tensor(self, value: torch.Tensor):
+                # Guards are added inside register_attr_or_module
+            )
+
+        if type(value) in config.traceable_tensor_subclasses:
+            # Ordinarily, we would fakeify a tensor so that it can get dynamic
+            # shapes and be computed on without triggering actual operations.
+            # However, how can we fakeify a tensor subclass?  Ordinary
+            # inheritance (nor multiple inheritance) won't work work.
+            #
+            # Instead, our plan is to *manually simulate* the tensor subclass
+            # inheriting from a fake tensor with dynamo.  This means our
+            # data representation for a tensor subclass will be a fake tensor
+            # + tensor subclass type + any extra data the subclass may have
+            # been storing on the tensor.  Because all Python accesses are
+            # mediated through TensorWithTFOverrideVariable, we can ensure
+            # that we dispatch differently, e.g., according to
+            # __torch_function__
+            #
+            # To simplify things for now, the __dict__ tracking bits haven't
+            # been implemented yet, but they can be added into this design at
+            # a later point in time.
+            ignore_subclass = True
         else:
-            if not is_constant_source(self.get_source()):
-                self.tx.output.graphargs.append(
-                    GraphArg(self.get_source(), value, False)
-                )
-            # Disable __torch_function__ to prevent cloning of `value` to hit
-            # us
-            with torch._C.DisableTorchFunction():
-                if is_constant_source(self.get_source()):
-                    return self.tx.output.register_attr_or_module(
-                        value,
-                        re.sub(r"[^a-zA-Z0-9]+", "_", self.name),
-                        source=None,
-                        # Guards are added inside register_attr_or_module
-                    )
-                tensor_variable = wrap_fx_proxy(
-                    tx=self.tx,
-                    proxy=self.tx.output.create_graph_input(
-                        re.sub(r"[^a-zA-Z0-9]+", "_", self.name), type(value)
-                    ),
-                    example_value=value,
-                    guards=self.make_guards(GuardBuilder.TENSOR_MATCH),
-                    should_specialize=self.tensor_should_specialize(),
-                )
-            if torch.overrides.has_torch_function_unary(value):
-                subclass_torch_function__func = value.__torch_function__.__func__
-                subclass_type = type(value)
-                return TensorWithTFOverrideVariable(
-                    tensor_variable,
-                    self.get_source(),
-                    subclass_torch_function__func,
-                    subclass_type,
-                )
-            return tensor_variable
+            assert type(value) in (torch.Tensor, torch.nn.Parameter)
```

Comment:
nit: this will fail for every unknown subclasses so we should have a nicer error message here as this will be quite user facing no?

Reply:
No, we test if it's a traceable tensor type earlier in dynamo, so if you have gotten here we've messed up the preconditions.